examples/jupyter-tutorials/API.ipynb

```diff
@@ -11,188 +11,1039 @@
+    "\n",
+    "# random_uniform\n",
+    "left = -1\n",
+    "right = 1\n",
+    "e = dy.random_uniform(dim, left=left, right=right, batch_size=batch_size)\n",
+    "print('A {} dim random uniform distribution of left={} and right={}, batch_size={}:\\n{}'.format(dim, left, right, batch_size, e.npvalue()))\n",
+    "\n",
+    "# random_gumbel\n",
+    "# Create a vector distributed according to a Gumbel distribution with the specified parameters. \n",
+    "# (Currently only the defaults of mu=0.0 and beta=1.0 supported.\n",
+    "mu = 0.0\n",
+    "beta = 1.0\n",
+    "e = dy.random_gumbel(dim, mu=mu, beta=beta, batch_size=batch_size)\n",
+    "print('A {} dim random gumbel distribution of mu={} and beta={}, batch_size={}:\\n{}'.format(dim, mu, beta, batch_size, e.npvalue()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "## ==== We can take the value of an expression. \n",
     "# For complex expressions, this will run forward propagation.\n",
-    "print z.value()    \n",
-    "print z.npvalue()      # as numpy array\n",
-    "print v.vec_value()    # as vector, if vector\n",
-    "print x.scalar_value() # as scalar, if scalar\n",
-    "print x.value()        # choose the correct one\n",
-    "\n",
-    "## ==== Parameters\n",
+    "print(mat1.value())   \n",
+    "print(mat1.npvalue())    # as numpy array\n",
+    "print(v.vec_value())     # as vector, if vector\n",
+    "print(x.scalar_value())  # as scalar, if scalar\n",
+    "print(x.value())         # choose the correct one"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create Parameters\n",
+    "Parameters are things need to be trained. In contrast to a system like Torch where computational modules may have their own parameters, in DyNet parameters are just parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "# Parameters are things we tune during training.\n",
     "# Usually a matrix or a vector.\n",
     "\n",
     "# First we create a parameter collection and add the parameters to it.\n",
-    "m = ParameterCollection() \n",
-    "pW = m.add_parameters((8,8)) # an 8x8 matrix\n",
-    "pb = m.add_parameters(8)\n",
-    "\n",
-    "# then we create an Expression out of the parameter collection's parameters\n",
-    "W = dy.parameter(pW)\n",
-    "b = dy.parameter(pb)\n",
-    "\n",
+    "m = dy.ParameterCollection() \n",
+    "W = m.add_parameters((8,8)) # an 8x8 matrix, return an expr\n",
+    "b = m.add_parameters(8) # an 8x1 vector, return as expr"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It should be noticed that in DyNet 2.0+ and later version, the dy.parameters() is depecated so explicitly adding parameters to the computation graph is no longer necessary. Any used parameter will be added automatically."
```

Comment:
It should be noticed that in DyNet 2.0+ and later version, the dy.parameters() is depecated
->
Previously, it was necessary to add parameters to the computation graph using "dy.parameters()", but this is now deprecated

Reply:
Also it should probably be 2.0.4+ or ">2.0.3"so that people don't get confused